content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,695 @@
+proposals arranged in a directed acyclic graph.  Nodes would naturally
+choose to compute the consensus of directed graph of the proposals
+that they are interested in.  This autosharding property superceeds
+any possible Nakomota consensus mechanism, as any proof of work
+mechanism would by defition need to increase its the work necessary
+for its continued constant security of all chains of interest.
+
+# Glacier Algorithm Specification
+
+The Glacier consensus algorithm computes a boolean decision on a
+proposition via a set of distributed computational nodes.  Glacier is
+a leaderless probabilistic binary consensus algorithm with fast
+finality that provides good reliability for network and Byzantine
+fault tolerance.
+
+## Algorithm 
+
+The algorithm begins with considering the truth status of a given
+proposition.
+
+A proposal is formulated to which consensus of truth or falsity is
+desired.  Each node that participates starts the protocol with an
+opinion on the proposal, represented in the sequel as `no`, `none`,
+and `yes`.
+
+A new proposition is discovered either by local creation or in
+response to a query, a node checks its local opinion.  If the node can
+compute a justification of the proposal, it sets its opinion to one of
+`yes` or `no`.  If it cannot form an opinion, it leaves its opinion as
+`none`.
+
+The node then participates in a number of query rounds in which it
+solicits other node's opinion in query rounds.  Given a set of `n`
+leaderless computational nodes, a gossip-based protocol is presumed to
+exist which allows members to discover, join, and leave a weakly
+transitory maximally connected graph.  Joining this graph allows each
+node to view a possibly incomplete node membership list of all other
+nodes.  This view may change as the protocol advances, as nodes join
+and leave.
+
+
+### Proposal Identification
+
+The node has a semantics and serialization of the proposal, of which
+it sets an initial opinion:
+
+     opinion
+        <-- initial opinion on truth of the proposal
+            as one of: {NO, NONE, YES}
+    
+The proposal proceeds in asynchronous rounds, in which each node
+queries `k` randomly sampled nodes for their opinions until a decision
+about local finality is achieved.  The liveness of the algorithm is
+severely constrained in the absence of timeouts for a round to
+proceed.  When a given node has finalized its decision on the
+proposal, it enters a quiescent state in which it optionally discards
+all information gathered during the query process retaining only the
+final opinion on the truth of the proposal.
+
+### Setup Parameters
+
+The node initializes the following integer ratios as constants:
+```
+;;; The following values are constants chosen with justification from experiments
+;;; performed with the adversarial models
+
+;; FIXME: find and justify an empirical confidence threshold 
+confidence_threshold
+  <-- 1  ;; BOGUS:  this should be a function of the network size
```

Comment:
Suppose that at time $t=0$ \emph{all}  $N$ nodes in the network  are  in state Yes (or No), and there are no adversaries,  then  each node in the  \emph{Glacier} will stop updating after $\frac{\ell c}{K(1-c)}$ iterations, where $c\in(0,1)$ is the confidence  threshold, $\ell$ is the look-ahead parameter  and $K$ is  the query size. So $c$ has to be strictly less that $1$, otherwise number of iterations will diverge as $c$ approaches $1$.